pkgs/development/compilers/llvm/11/compiler-rt/default.nix

```
@@ -3,7 +3,6 @@
 let
 
   useLLVM = stdenv.hostPlatform.useLLVM or false;
-  isDarwin = stdenv.hostPlatform.isDarwin;
   bareMetal = stdenv.hostPlatform.parsed.kernel.name == "none";
```

What if you did?
```suggestion
  isNewDarwinBootstrap = stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64;
  bareMetal = stdenv.hostPlatform.parsed.kernel.name == "none";
```
and then `isDarwin -> `isNewDarwinBootstrap` everywhere below? I'm excited about just being able to use `useLLVM` on darwin with a `linker = "cctools"` override, but what you wrote below I think makes us a bit farther from that?

To be extra clear, I *thtink* you can also do
```
  # What I believe you called it for LLVM 7
  standalone = useLLVM || isNewDarwinBootstrap;
```
and use that to rewrite every usage of `useLLVM` and `isNewDarwinBootstrap` below?

---

I've pushed an approximation of this with a couple of changes: On darwin we always `haveLibc` so I omitted `isNewDarwinBootstrap` from the branches it will never affect. Secondly, and perhaps I misunderstand, but the sanitizers/xray/libfuzzer/profile features should work on Darwin, so removing them as part of `isNewDarwinBootstrap` seems wrong. I don't know why it doesn't build for aarch64-darwin though.

---

Oh you bring up some god points. I'll make some suggestions base on that.